### PR TITLE
Add a document to describe which log level to use

### PR DESCRIPTION
**Pull request by ahoppen**
Also change a few log levels and make all log messages consistently start with an uppercase letter.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

File: Documentation/Logging.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Logging
+
+The following is a guide of how to write log messages and which level they should be logged at. There may be reasons to deviate from these guides but they should be a good starting point.
+
+## Log Levels
+
+The following log levels should be used by log messages in default. The [Explore logging in Swift](https://developer.apple.com/wwdc20/10168?time=604) session from WWDC20 has some explanation of how these levels are persisted by OS log and we follow those guidelines.
```
Comment (hamishknight) [inline Documentation/Logging.md:7]: Not sure I understand `should be used by log messages in default`

File: Documentation/Logging.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+> From Explore Logging in Swift: _Error seen during execution_
+
+An error that is due to user input or eg. stale state of files on disk. It indicates that something is going wrong which might explain unexpected behavior. Errors could be due to malformed user input such as invalid requests from the editor or due to stale state that will eventually converge again.
+
+Examples:
+- The client sends an invalid request
+- Preparation of a file failed due to a syntax error in the user’s code
+- The index contains a reference to a source file but the source fail has been modified since the index was last updated and is thus no longer valid
+
+## Log/Notice/Default
```
Comment (hamishknight) [inline Documentation/Logging.md:31]: Is this (and the following headers) intentionally H2 instead of H3?

File: Documentation/Logging.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- The user sends an LSP request
+- Indexing of a file starts or finishes
+- New build settings for a file have been computed
+- Responses from sourcekitd
+
+## Info
+
+> From Explore Logging in Swift: _Helpful but not essential for troubleshooting_ (not persisted, logged to memory)
+
+Internal state transitions that are helpful. If eg. a request fails and it’s not immediately obvious at which it failed, these should help us narrow down the code that it failed in. These messages might be missing from the logs generated by `sourcekit-lsp diagnose` and should not generally be needed to fix issues
```
Comment (hamishknight) [inline Documentation/Logging.md:49]:
```suggestion
Internal state transitions that are helpful. If eg. a request fails and it’s not immediately obvious at which point it failed, these should help us narrow down the code that it failed in. These messages might be missing from the logs generated by `sourcekit-lsp diagnose` and should not generally be needed to fix issues
```

**Pull request merged**